src/main/java/io/lakefs/routerfs/RouterFileSystem.java

```diff
@@ -252,4 +300,12 @@ private Path createSchemedPath(Path p) {
         }
         return p;
     }
+
+    private FileStatus convertFileStatusToSrcPath(FileStatus fileStatus, PathProperties pathProperties) {
```

Comment:
This function seems misnamed.  AFAICS it translates the path on a FileStatus, it doesn't convert a FileStatus to any kind of Path.